pkg/urbit/vere/io/term.c

```diff
@@ -358,87 +322,91 @@ _term_write_cb(uv_write_t* wri_u, c3_i sas_i)
   c3_free(wri_u);
 }
 
-/* _term_it_write_buf(): write buffer uv style.
+/* _term_it_write(): write libuv buffer, freeing pointer.
 */
 static void
-_term_it_write_buf(u3_utty* uty_u, uv_buf_t buf_u)
+_term_it_write(u3_utty*  uty_u,
+               uv_buf_t* buf_u,
+               void*     ptr_v)
 {
-  uv_write_t* wri_u = c3_malloc(sizeof(uv_write_t));
-  wri_u->data = buf_u.base;
-
-  c3_w ret_w;
-  if ( 0 != (ret_w = uv_write(wri_u,
-                     (uv_stream_t*)&(uty_u->pop_u),
-                     &buf_u, 1,
-                     _term_write_cb)) )
+  uv_write_t* wri_u = c3_malloc(sizeof(*wri_u));
+  c3_w        ret_w;
+
+  wri_u->data = ptr_v;
+
+  if ( (ret_w = uv_write(wri_u,
+                         (uv_stream_t*)&(uty_u->pop_u),
+                         buf_u, 1,
+                         _term_write_cb)) )
   {
     u3l_log("term: write: %s\n", uv_strerror(ret_w));
   }
 }
 
-/* _term_it_write_old(): write buffer, transferring pointer.
+/* _term_it_dump(): write static vector.
 */
 static void
-_term_it_write_old(u3_utty* uty_u,
-                   u3_ubuf* old_u)
+_term_it_dump(u3_utty*    uty_u,
+              c3_w        len_w,
+              const c3_y* hun_y)
 {
-  uv_buf_t buf_u;
-
-  //  XX extra copy here due to old code.  Use hbod as base directly.
-  //
-  {
-    c3_y* buf_y = c3_malloc(old_u->len_w);
-
-    memcpy(buf_y, old_u->hun_y, old_u->len_w);
-    buf_u = uv_buf_init((c3_c*)buf_y, old_u->len_w);
-
-    c3_free(old_u);
-  }
-  _term_it_write_buf(uty_u, buf_u);
+  uv_buf_t buf_u = uv_buf_init((c3_c*)hun_y, len_w);
+  _term_it_write(uty_u, &buf_u, 0);
```

Comment:
You could consider using `_term_it_dump_buf` here for consistency. You use it (instead of manually passing a `0` pointer) everywhere else.